From 5180f41618df6a8f3c38c6e7c122d37c26ef39dd Mon Sep 17 00:00:00 2001
From: He Junyan <junyan.he@intel.com>
Date: Tue, 16 Apr 2024 18:13:06 +0800
Subject: [PATCH 059/117] va: baseenc: Set the trellis parameter anyway

The driver may enable trellis by default. So we should also set the
trellis info to driver even when the trellis option is turned off.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/6663>
---
 sys/va/gstvabaseenc.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/sys/va/gstvabaseenc.c b/sys/va/gstvabaseenc.c
index 0011f7172..a2a9b84f0 100644
--- a/sys/va/gstvabaseenc.c
+++ b/sys/va/gstvabaseenc.c
@@ -1169,17 +1169,14 @@ gst_va_base_enc_add_trellis_parameter (GstVaBaseEnc * base,
   } trellis = {
     .type = VAEncMiscParameterTypeQuantization,
     .tr.quantization_flags.bits = {
-       .disable_trellis = 0,
-       .enable_trellis_I = 1,
-       .enable_trellis_B = 1,
-       .enable_trellis_P = 1,
+       .disable_trellis = !use_trellis,
+       .enable_trellis_I = use_trellis,
+       .enable_trellis_B = use_trellis,
+       .enable_trellis_P = use_trellis,
     },
   };
   /* *INDENT-ON* */
 
-  if (!use_trellis)
-    return TRUE;
-
   if (!gst_va_encoder_add_param (base->encoder, picture,
           VAEncMiscParameterBufferType, &trellis, sizeof (trellis))) {
     GST_ERROR_OBJECT (base, "Failed to create the trellis parameter");
-- 
2.43.0

